### PR TITLE
Allow programs to be filtered by comma-separated UUIDs

### PR DESCRIPTION
**Pull request by rlucioni**
ECOM-5261

@edx/ecommerce please review. @AlasdairSwan FYI.

### CODE REVIEW

File: course_discovery/apps/api/v1/tests/test_views/test_programs.py [hunk 1]
```diff
@@ -54,3 +54,16 @@ def test_filter_by_type(self):
 
         response = self.client.get(url + 'bar')
         self.assertEqual(response.data['results'], [])
+
+    def test_filter_by_uuids(self):
+        """ Verify that the endpoint filters programs to those matching the provided UUIDs. """
+        url = reverse('api:v1:program-list') + '?uuids='
```
Comment (clintonb) [inline course_discovery/apps/api/v1/tests/test_views/test_programs.py]: Minor: Move `reverse('api:v1:program-list')` to a class variable (e.g. `list_path`) to DRY up this class.

**Comment by clintonb** [comment]
@rlucioni any idea if casing of the UUID matters?

👍

File: course_discovery/apps/api/v1/tests/test_views/test_programs.py (hunk 1 above)
Comment (rlucioni) [inline course_discovery/apps/api/v1/tests/test_views/test_programs.py]: Good call, will update.

**Comment by rlucioni** [comment]
@clintonb I suspect it may, will experiment and see what I can do to make it case-insensitive. Good catch.

**Comment by rlucioni** [comment]
@clintonb lookups on UUIDFields appear to be case-insensitive:

``` py
>>> from course_discovery.apps.course_metadata.models import Program
>>> Program.objects.all()
[<Program: foo>]
>>> Program.objects.get().uuid
UUID('a4dc1540-f75e-4637-ae53-be4850593510')
>>> Program.objects.filter(uuid='a4dc1540-f75e-4637-ae53-be4850593510')
[<Program: foo>]
>>> Program.objects.filter(uuid='a4dc1540-f75e-4637-ae53-be4850593510'.upper())
[<Program: foo>]
```

**Pull request merged**